enh(plugin) add metricnamespace option to Azure Monitor get-metric mode

petneli opened the pull request:
Added metricnamespace as parameter to the Azure Metrics script

Fixes #2248

Sims24 reviewed (commented):
Hi @petneli 

Thanks for your contribution. It's cool to have a new contributor, welcome :)

Few questions/remarks: 

- Could you share what do you plan to do with the 'Count' aggregation? I'm interested in the usecase behind. We didn't add it as it returns the number of records but doesn't deal with the measurement itself.
- Could you also update the azcli.pm custommode? We try to have same feature in both.

Simon

Sims24 commented on cloud/azure/management/monitor/mode/getmetrics.pm:
```diff
@@ -230,6 +235,10 @@ Set resource type (Required if resource's name is used).
 
 Set monitor metrics (Required) (Can be multiple).
 
+=item B<--metricnamespace>
```
Tiny detail, could you name it --metric-namespace with a dash as word separator ;)

petneli commented:
Hi @Sims24,

- We have defined custom metric with aggregation "count" and couldn't monitor it + it is in general supported by Azure, so I thought it is better to have it then not. Interesting use cases would be e.g. to see the number of data points in case of missing data (see "More complex example" [metrics-aggregation-explained](https://docs.microsoft.com/en-us/azure/azure-monitor/essentials/metrics-aggregation-explained) )
- I have adjusted azcli.pm as well. ( added "count" as aggregation and "metric namespace" parameter)
- I have renamed the parameter "--metricnamespace" to "--metric-namespace" as to comply with the general style.

Neli

Sims24 commented:
Wonderful!

Thank you
